src/layman/layer/rest_workspace_layers.py

```diff
@@ -82,17 +76,10 @@ def post(workspace):
     check_crs = crs_id is None
 
     # FILE NAMES
-    if use_chunk_upload and zipped_file:
-        filenames = list()
-    elif use_chunk_upload:
-        filenames = files
-    elif zipped_file:
-        filenames = fs_util.get_filenames_from_zip_storage(zipped_file, with_zip_in_path=True)
-    else:
-        filenames = [f.filename for f in files]
-    file_type = input_file.get_file_type(input_file.get_main_file_name(filenames))
-    if not (use_chunk_upload and zipped_file):
-        input_file.check_filenames(workspace, layername, filenames, check_crs)
+    use_chunk_upload = not input_files.sent_streams
+    if not (use_chunk_upload and input_files.is_one_archive):
+        input_file.check_filenames(workspace, layername, input_files, check_crs)
+    file_type = input_file.get_file_type(input_file.get_main_file_name(input_files.raw_or_archived_paths))
```

Comment:
I think, you can use directly `input_files.raw_or_archived_main_file_path` instead of `input_file.get_main_file_name(input_files.raw_or_archived_paths)`